src/controllers/mainController.ts

```diff
@@ -930,10 +930,22 @@ export default class MainController implements vscode.Disposable {
             for (let conn of newConnections) {
                 // if a connection is not connected
                 // that means it was added manually
-                const uri = ObjectExplorerUtils.getNodeUriFromProfile(<IConnectionProfile>conn);
+                const newConnectionProfile = <IConnectionProfile>conn;
+                const uri = ObjectExplorerUtils.getNodeUriFromProfile(newConnectionProfile);
                 if (!this.connectionManager.isActiveConnection(conn) &&
                     !this.connectionManager.isConnecting(uri)) {
-                    // add a disconnected node for it
+                    // if the added connection is SQL Auth and has password
+                    // save the password and remove it from settings
+                    if (conn.authenticationType === Constants.sqlAuthentication &&
+                        !Utils.isEmpty(conn.password)) {
+                        await this.connectionManager.connectionStore.saveProfile(newConnectionProfile);
+                        const newConnectionProfileCopy = Object.assign({}, newConnectionProfile);
+                        newConnectionProfileCopy.password = '';
+                        this._vscodeWrapper.setConfiguration(Constants.extensionConfigSectionName,
```

Comment:
Isn't this going to overwrite the entire array of connections with just the single one?

Reply:
No, it actually updates the existing configuration instead of replacing it

Reply:
What do you mean "updates"? Isn't the configuration in this case an array of profiles? How does it know which one to "update"?

Reply:
https://github.com/microsoft/vscode-mssql/pull/1512/files#diff-59d53de3ec5d967a3079fee08a1391d7R345

Reply:
https://code.visualstudio.com/api/references/vscode-api#WorkspaceConfiguration

Reply:
Yes. So first you get the configuration under the `mssql` section with the id `connections`. This should be an array of profiles
![image](https://user-images.githubusercontent.com/28519865/70083111-a5ee8d80-15c0-11ea-81c2-10e4f3804446.png)

 And then you update the value of that `connections` config and set it to a single profile - the new one you cloned here. Am I missing something?

It should be simple to test. Have multiple profiles saved, then add the password field to one of them such that it runs through this logic. What does the config look like afterwards? 

If you say you've tested this and it works as expected I'll take your word for it - but my understanding of what is happening here doesn't line up with what you're saying.

Reply:
Yes, I tested with multiple connections already. From what I know and have seen, the update method updates the section with the new object. If you want to get rid of any object, you need to explicitly update with `undefined` - 

```
Note 2: To remove a configuration value use undefined, like so: config.update('somekey', undefined)
```

Reply:
Especially since you're doing this kind of logic already but there you're calling update with an array of profiles instead.

https://github.com/Microsoft/vscode-mssql/blob/2e745d3d24344216b453c9f8db334775a9bae2a4/src/connectionconfig/connectionconfig.ts#L153
```
private writeProfilesToSettings(profiles: IConnectionProfile[]): Promise<void> {
        // Save the file
        const self = this;
        return new Promise<void>((resolve, reject) => {
            self._vscodeWrapper.getConfiguration(Constants.extensionName).update(Constants.connectionsArrayName, profiles, true).then(() => {
                resolve();
            }, err => {
                reject(err);
            });
        });
    }
```

Reply:
I'll update the function just to be safe, though I tested with multiple profiles